lua/lspconfig/server_configurations/solidity_ls_nomicfoundation.lua

```diff
@@ -1,21 +1,33 @@
 local util = require 'lspconfig.util'
 
+local root_files = {
+  'hardhat.config.js',
+  'hardhat.config.ts',
+  'foundry.toml',
+  'remappings.txt',
+  'truffle.js',
+  'truffle-config.js',
+  'ape-config.yaml',
+}
+
 return {
   default_config = {
     cmd = { 'nomicfoundation-solidity-language-server', '--stdio' },
     filetypes = { 'solidity' },
-    root_dir = util.root_pattern('.git', 'package.json'),
+    root_dir = util.root_pattern(unpack(root_files)) or util.root_pattern('.git', 'package.json'),
+    single_file_support = true,
   },
   docs = {
     description = [[
 https://github.com/NomicFoundation/hardhat-vscode/blob/development/server/README.md
 
+`nomicfoundation-solidity-language-server` can be installed via `npm`:
+
+```sh
 npm install -g @ignored/solidity-language-server
+```
 
 A language server for the Solidity programming language, built by the Nomic Foundation for the Ethereum community.
 ]],
-    default_config = {
```

Comment:
why remove these lines ?

Reply:
It would be too long to explain with all the `root_files` so I just followed the example of other LSPs (`git grep "see source file"`) and removed it.

Diff to docs generated by `scripts/docgen.sh`:

````diff
diff --git a/doc/server_configurations.md b/doc/server_configurations.md
index 63dd765..aeb5441 100644
--- a/doc/server_configurations.md
+++ b/doc/server_configurations.md
@@ -8572,7 +8572,11 @@ require'lspconfig'.solidity_ls.setup{}
 
 https://github.com/NomicFoundation/hardhat-vscode/blob/development/server/README.md
 
+`nomicfoundation-solidity-language-server` can be installed via `npm`:
+
+```sh
 npm install -g @ignored/solidity-language-server
+```
 
 A language server for the Solidity programming language, built by the Nomic Foundation for the Ethereum community.
 
@@ -8595,7 +8599,11 @@ require'lspconfig'.solidity_ls_nomicfoundation.setup{}
   ```
   - `root_dir` : 
   ```lua
-  root_pattern(".git", "package.json")
+  see source file
+  ```
+  - `single_file_support` : 
+  ```lua
+  true
   ```
 
 
diff --git a/doc/server_configurations.txt b/doc/server_configurations.txt
index 63dd765..aeb5441 100644
--- a/doc/server_configurations.txt
+++ b/doc/server_configurations.txt
@@ -8572,7 +8572,11 @@ require'lspconfig'.solidity_ls.setup{}
 
 https://github.com/NomicFoundation/hardhat-vscode/blob/development/server/README.md
 
+`nomicfoundation-solidity-language-server` can be installed via `npm`:
+
+```sh
 npm install -g @ignored/solidity-language-server
+```
 
 A language server for the Solidity programming language, built by the Nomic Foundation for the Ethereum community.
 
@@ -8595,7 +8599,11 @@ require'lspconfig'.solidity_ls_nomicfoundation.setup{}
   ```
   - `root_dir` : 
   ```lua
-  root_pattern(".git", "package.json")
+  see source file
+  ```
+  - `single_file_support` : 
+  ```lua
+  true
   ````
 
 
```

Reply:
@glepnir Did I answer the question? Or I can document as `[[root_pattern('.git', 'package.json' , 'hardhat.config.js', 'hardhat.config.ts', 'foundry.toml', 'remappings.txt', 'truffle.js', 'truffle-config.js', 'ape-config.yaml')]]`
 if preferred

Reply:
@glepnir I figured out how to put the docs back. The doc string is dynamically generated now.